oci/spec_opts.go

```diff
@@ -1633,7 +1633,7 @@ func WithWindowsNetworkNamespace(ns string) SpecOpts {
 }
 
 // WithCDIDevices injects the requested CDI devices into the OCI specification.
-func WithCDIDevices(devices ...string) SpecOpts {
+func WithCDIDevices(devices, mounts []string) SpecOpts {
```

Comment:
what "mounts" represent here (from the linked issues) are *optional* CDI devices. Refering to them as mounts is incorrect.

If this is something that we want to enable from a Containerd perspective I would suggest handling them separately. In addition, I would like to see some attention paid to how we would allow a user to specify that a CDI device is to be considered optional. This should ideally also be compatible with other mechanisms for interacting with the spec including CLI tools such as `ctr`.

Reply:
What I mean by handling these separately is to introduce a `WithOptionalCDIDevices` that can be used to process the devices associated with the CSI mounts.

Reply:
Thx @elezar I have updated the code. and introduce a new withCDOpts for handling containerMounts generated in CSI.